buildSrc/src/main/kotlin/io/spine/internal/gradle/javascript/task/Assemble.kt

```diff
@@ -0,0 +1,181 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.javascript.task
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import io.spine.internal.gradle.base.assemble
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+
+/**
+ * Registers tasks for assembling JavaScript artifacts.
+ *
+ * List of tasks to be created:
+ *
+ *  1. [TaskContainer.assembleJs];
+ *  2. [TaskContainer.compileProtoToJs];
+ *  3. [TaskContainer.installNodePackages];
+ *  4. [TaskContainer.updatePackageVersion].
+ *
+ * An example of how to apply it in `build.gradle.kts`:
+ *
+ * ```
+ * import io.spine.internal.gradle.javascript.javascript
+ * import io.spine.internal.gradle.javascript.task.assemble
+ *
+ * // ...
+ *
+ * javascript {
+ *     tasks {
+ *         assemble()
+ *     }
+ * }
+ * ```
+ */
+fun JsTasks.assemble() {
+
+    installNodePackages()
+    compileProtoToJs()
+    updatePackageVersion()
+
+    assembleJs().also {
+        assemble.configure {
+            dependsOn(it)
+        }
+    }
+
+}
+
+
+/**
+ * Locates `assembleJs` task in this [TaskContainer].
+ *
+ * It is a lifecycle task that produces consumable JavaScript artifacts.
+ */
+val TaskContainer.assembleJs: TaskProvider<Task>
+    get() = named("assembleJs")
+
+private fun JsTasks.assembleJs() =
+    register("assembleJs") {
+
+        description = "Assembles JavaScript sources into consumable artifacts."
+        group = jsAssembleTask
+
+        dependsOn(
+            installNodePackages,
+            compileProtoToJs,
+            updatePackageVersion
+        )
+    }
+
+
+/**
+ * Locates `compileProtoToJs` task in this [TaskContainer].
+ *
+ * The task compiles Protobuf messages into JavaScript.
+ *
+ * This is a lifecycle task that performs no action itself. It is used to aggregate other tasks
+ * which perform the compilation. Those tasks are to be provided by `Protobuf`
+ * and `McJsPlugin` plugins.
+ */
+val TaskContainer.compileProtoToJs: TaskProvider<Task>
+    get() = named("compileProtoToJs")
+
+private fun JsTasks.compileProtoToJs() =
+    register("compileProtoToJs") {
+
+        description = "Compiles Protobuf messages into JavaScript."
+        group = jsAssembleTask
+    }
+
+
+/**
+ * Locates `installNodePackages` task in this [TaskContainer].
+ *
+ * The task installs a package and any packages that it depends on using the `npm install` command.
```

Comment:
I don't understand this part: "installs a package and any packages that it depends on". There is no certain "package" in this context. Therefore, there's nothing to speak of.

Please rephrase.